components/places/src/history_sync/store.rs

```diff
@@ -138,6 +138,7 @@ impl<'a> HistoryStore<'a> {
         mem_cached_state: &mut MemoryCachedState,
         disk_cached_state: &mut Option<String>,
         sync_ping: &mut telemetry::SyncTelemetryPing,
+        service_status: &mut ServiceStatus,
```

Comment:
Mmmm...on the one hand, I'm really not a fan of mutable out parameters, especially when this function has a `()` return result. On the other, it does make returning _errors_ more complicated, so it's probably fine. 😅

Reply:
Your suggestion re SyncResult really cleaned that up, thanks!